modules/standard/Path.chpl

```diff
@@ -62,18 +62,82 @@
 */
 module Path {
 
+
 use SysError;
 use Sys;
 
-/* Represents generally the current directory.  This starts as the directory
-   where the program is being executed from.
- */
+
+/*
+  Represents generally the current directory. This starts as the directory
+  where the program is being executed from.
+*/
 const curDir = ".";
 /* Represents generally the parent directory. */
 const parentDir = "..";
 /* Denotes the separator between a directory and its child. */
 const pathSep = "/";
 
+
+/*
+  Normalize a path by eliminating redundant separators and up-level references.
+  The paths ``foo//bar``, ``foo/bar/``, ``foo/./bar``, and ``foo/baz/../bar``
+  would all be changed to ``foo/bar``.
+
+  .. warning::
+
+    May alter meaning of paths containing symbolic links.
+
+  .. note::
+
+    Unlike its Python counterpart, this function does not (currently) change
+    slashes to backslashes on Windows.
+
+  :arg name: a potential path to collapse, possibly destroying the meaning of
+             the path if symbolic links were included.
+  :type name: `string`
+
+  :return: the collapsed version of `name`
+  :rtype: `string`
+*/
+proc normPath(name: string): string {
+
+  // Python 3.7 implementation:
+  // https://github.com/python/cpython/blob/3.7/Lib/posixpath.py
+
+  if name == "" then
+    return curDir;
+
+  var leadingSlashes = if name.startsWith(pathSep) then 1 else 0;
+
+  // Two leading slashes has a special meaning in POSIX.
+  if name.startsWith(pathSep * 2) && !name.startsWith(pathSep * 3) then
+    leadingSlashes = 2;
```

Comment:
I actually originally had a helper function that did that, but changed it when I realized that a malignant input of all leading slashes was O(n).

Would you still like me to move this prelude code into a helper function?

Reply:
You could always have it stop after 3 or the end of the string, whichever comes first.  I think it would be good to avoid hard-coding those numbers in and manually updating them (not that I expect the implementation to change over time), but it doesn't have to be in a helper function.